context-propagation/src/main/java/io/micrometer/context/ContextRegistry.java

```diff
@@ -145,6 +145,36 @@ public ContextRegistry registerThreadLocalAccessor(ThreadLocalAccessor<?> access
         return this;
     }
 
+    /**
+     * Removes a {@link ThreadLocalAccessor}.
+     * @param key under which the accessor got registered
+     * @return the same registry instance
+     */
+    public ContextRegistry removeThreadLocalAccessor(String key) {
+        for (ThreadLocalAccessor<?> existing : this.threadLocalAccessors) {
+            if (existing.key().equals(key)) {
+                this.threadLocalAccessors.remove(existing);
+                break;
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Removes a {@link ContextAccessor} of a given type.
+     * @param accessorToRemove class of {@link ContextAccessor} to remove
+     * @return the same registry instance
+     */
+    public ContextRegistry removeContextAccessor(Class<? extends ContextAccessor<?, ?>> accessorToRemove) {
+        for (ContextAccessor<?, ?> existing : this.contextAccessors) {
+            if (accessorToRemove.isAssignableFrom(existing.getClass())) {
```

Comment:
even if there are protections about adding a `ContextAccessor` with similar readable/writeable types, I would be careful here just in case a more generic `class` is provided which would remove a more specialized instance (think `Number.class.isAssignableFrom(Double.class)`)

since the API allows to register a `ContextAccessor` by providing an instance, why not simply use the reverse API: provide the instance that you want to remove (same as `List#remove(Object)`)? That way you have a mean of reversing a manual registration. The service-loaded accessors wouldn't be removable, but I think that is fine.

Reply:
Fair enough, the SPI loaded accessor most likely shouldn't be removed anyways